modules/hardware/console.nix

```diff
@@ -0,0 +1,80 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.hardware.console;
+
+  mkBinaryKeyMap = km: pkgs.runCommand "bkeymap"
+    {
+      nativeBuildInputs = [ pkgs.buildPackages.kbd ];
+      preferLocalBuild = true;
+    } ''
+      loadkeys --bkeymap "${km}" >$out
+    '';
+in
+{
+  options = {
+    hardware.console = {
+      enable = lib.mkOption {
+        description = "Whether to configure the console at boot.";
+        type = lib.types.bool;
+        default = true;
+      };
+
+      setvesablank = lib.mkOption {
+        description = "Turn VESA screen blanking on or off.";
+        type = lib.types.bool;
+        default = true;
+      };
+
+      keyMap = lib.mkOption {
+        description = ''
+          The keyboard mapping table for the virtual consoles.
+          This option may have no effect if
+          hardware.console.binaryKeyMap is set.
+        '';
+        type = with lib.types; either str path;
+        example = "us";
+        # Not setting a default here can be obnoxious but
+        # not worse than favoring a culturally biased
+        # layout optimised for mechanical typewriters.
+        # We should localize the templates and auto-detected
+        # configs instead of the defaults.
```

Comment:
i think ultimately you want a default here... but yeah, what you said about auto-detecting sounds good because it becomes a runtime thing :+1:

Reply:
I'm being obnoxious here but we can put one in later. Or provide localization config bundles.